nav2_collision_monitor/test/collision_monitor_node_test.cpp

```diff
@@ -1170,6 +1200,42 @@ TEST_F(Tester, testSourcesNotSet)
   cm_->cant_configure();
 }
 
+TEST_F(Tester, testCollisionPointsMarkers)
+{
+  rclcpp::Time curr_time = cm_->now();
+
+  // Set Collision Monitor parameters.
+  // Making two polygons: outer polygon for slowdown and inner for robot stop.
+  setCommonParameters();
+  addPolygon("Limit", POLYGON, 3.0, "limit");
+  addPolygon("SlowDown", POLYGON, 2.0, "slowdown");
+  addPolygon("Stop", POLYGON, 1.0, "stop");
+  addSource(SCAN_NAME, SCAN);
+  addSource(POINTCLOUD_NAME, POINTCLOUD);
+  addSource(RANGE_NAME, RANGE);
+  setVectors({"Limit", "SlowDown", "Stop"}, {SCAN_NAME, POINTCLOUD_NAME, RANGE_NAME});
+
+  // Start Collision Monitor node
+  cm_->start();
+
+  // Share TF
+  sendTransforms(curr_time);
+
+  // 1. Obstacle is far away from robot
+  publishCmdVel(0.5, 0.2, 0.1);
+  ASSERT_TRUE(waitCollisionPointsMarker(500ms));
+  ASSERT_EQ(collision_points_marker_msg_->markers[0].points.size(), 0u);
+
+  publishCmdVel(0.5, 0.2, 0.1);
+  publishScan(0.5, curr_time);
+  ASSERT_TRUE(waitData(0.5, 500ms, curr_time));
+  
+  ASSERT_TRUE(waitCollisionPointsMarker(500ms));
+  ASSERT_NE(collision_points_marker_msg_->markers[0].points.size(), 0u);
```

Comment:
fails but not sure why yet

Reply:
The points are empty... not sure, you should check!